packages/headless/src/features/facets/range-facets/date-facet-set/date-facet-set-slice.ts

```diff
@@ -59,6 +60,16 @@ export const dateFacetSetReducer = createReducer(
       .addCase(updateDateFacetSortCriterion, (state, action) => {
         handleFacetSortCriterionUpdate<DateFacetRequest>(state, action.payload);
       })
+      .addCase(updateDateFacetRangeAlgorithm, (state, action) => {
+        const {facetId, rangeAlgorithm} = action.payload;
+        const facetRequest = state[facetId];
+
+        if (!facetRequest) {
+          return;
+        }
+
+        facetRequest.rangeAlgorithm = rangeAlgorithm;
```

Comment:
It would be better to extract the logic into a `handRangeAlgorithmUpdate` function and reuse it here and for the numeric-slice.

It would also be good to add a test for the case where the facetRequest does not exist.